File: Unidad 4-Arreglos/ejercicio1.py
import random
#Autor: Jose Eduardo Mendez Verdejo    10/Marzo/19
#Entradas: una matriz de 15 x 12 elementos numericos
#Salidas: el numero menor de la matriz, la suma de sus primeras 5 filas y la suma de las 5 ultimas columnas
#Proceso: se obtiene una matriz de 15 x 12, se busca el elementos menor, se suman las 5 primeras filas y las
#		  ultimas 5 columnas.


#----- Se obtiene la matriz -----
def generarMatriz(matriz):

	for i in range(0,15):
		for j in range(0,12):

			#se generan los elementos de forma aleatoria en un rango de 0 a 999
			matriz[i][j] = random.randrange(0,1000) #para ingreso manual " int(input("ingrese un numero: ")) "

#----- Se busca el elemento menor -----
def menorMatriz(matriz):
	trono = 99999 #se asiga a una variable un valor mas grande para realizar la comparacion
	
	for i in range(0,15):
		for j in range(0,12):
			
			#si un numero es menor al trono se remplaza
			if(matriz[i][j] < trono):
				trono = matriz[i][j]

	return trono			

#----- Se suman las 5 filas -----
def sumaFilas(matriz,n):

	sumador = 0

	for i in range(0,5):
		for j in range(0,n):
			sumador += matriz[i][j]

	return sumador			

#----- Se suman las 5 columnas -----
def sumaColumnas(matriz,n):
	
	sumador = 0

	for i in range(0,15):
		for j in range(n-5,n):
			sumador += matriz[i][j]

	return sumador	


matriz = []
n = 12

for i in range (0,15):
	matriz.append([0] * n)


generarMatriz(matriz)
menor = menorMatriz(matriz)
sumaF = sumaFilas(matriz,n)
sumaC = sumaColumnas(matriz,n)

#se imprime la matriz para verificar datos
for i in range(0,15):
	print(matriz[i])

print("numero menor es: ",menor)
print("suma total de las 5 filas es: ",sumaF)
print("suma total de las ultimas 5 columnas es: ",sumaC)		

File: Unidad 4-Arreglos/Ejercicio1.go
//1. Escribe un programa que lea una matriz de 15 por 12 y que calcule:
//  a. El menor elemento de la matriz.
//  b. La suma de los elementos que estén en las primeras 5 filas
//  c. La suma de los elementos que estén en las últimas 5 columnas.
//Autor: Rodrigo Hernandez Gongora
package main
import "fmt"

func leerMatriz(matriz *[15][12] int){
	var value int
	for i := 0; i < 15; i++ {
		for j := 0; j < 12; j++ {
			fmt.Printf("Introduce un numero\n")
			fmt.Scanf("%d", &value)
			matriz[i][j]=value
		}
	}
}

func menor(matriz [15][12] int ) int {
// 32767 es el numero mas grande que se le puede asignar a un Int,
// por lo que se usa como bandera para un numero menor
	var min=32767 
	for i := 0; i < 15; i++ {
		for j := 0; j < 12; j++ {
			if matriz[i][j]<min {
				min=matriz[i][j]
			}
		}
	}
	return min
}

func suma1(matriz [15][12] int) int {
	var sum1=0 
	for i := 0; i < 5; i++ {
		for j := 0; j < 12; j++ {
			sum1+=matriz[i][j]
		}
	}
	return sum1
}

func suma2(matriz [15][12] int) int {
	var sum2=0 
	for i := 0; i < 15; i++ {
		for j := 8; j < 12; j++ {
			sum2+=matriz[i][j]
		}
	}
	return sum2
}

func salidas(x int, y int, z int) {
	fmt.Printf("El numero menor del arreglo es: %d\n", x);
    fmt.Printf("La suma de las primeras 5 filas es: %d\n", y);
    fmt.Printf("La suma de las ultimas 5 columnas es: %d\n", z);
}

func main() {
	var matriz[15][12] int
	var min int
	var sum1 int
	var sum2 int
	 leerMatriz(&matriz)
	 fmt.Printf("%d\n", matriz[0][0])
	 min=menor(matriz)
	 sum1=suma1(matriz)
	 sum2=suma2(matriz)
	 salidas(min, sum1, sum2)
}